samples/bluetooth/direct_test_mode/src/fem/nrf21540.c

```diff
@@ -514,7 +544,206 @@ int nrf21540_antenna_select(enum nrf21540_ant ant)
 
 	return err;
 }
+#if defined(NRF5340_XXAA_NETWORK)
+static inline nrf_spim_frequency_t get_nrf_spim_frequency(uint32_t frequency)
+{
+	/* Get the highest supported frequency not exceeding the requested one.
+	 */
+	if (frequency < 250000) {
+		return NRF_SPIM_FREQ_125K;
+	} else if (frequency < 500000) {
+		return NRF_SPIM_FREQ_250K;
+	} else if (frequency < 1000000) {
+		return NRF_SPIM_FREQ_500K;
+	} else if (frequency < 2000000) {
+		return NRF_SPIM_FREQ_1M;
+	} else if (frequency < 4000000) {
+		return NRF_SPIM_FREQ_2M;
+	} else if (frequency < 8000000) {
+		return NRF_SPIM_FREQ_4M;
+	} else {
+		return NRF_SPIM_FREQ_8M;
+	}
+}
+
+static void uarte_events_clear(NRF_UARTE_Type *uarte)
+{
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_ENDRX);
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_CTS);
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_NCTS);
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_RXDRDY);
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_TXDRDY);
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_ENDTX);
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_ERROR);
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_RXTO);
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_RXSTARTED);
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_TXSTARTED);
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_TXSTOPPED);
+}
+
+static void spim_events_clear(NRF_SPIM_Type *spim)
+{
+	nrf_spim_event_clear(spim, NRF_SPIM_EVENT_STOPPED);
+	nrf_spim_event_clear(spim, NRF_SPIM_EVENT_ENDRX);
+	nrf_spim_event_clear(spim, NRF_SPIM_EVENT_END);
+	nrf_spim_event_clear(spim, NRF_SPIM_EVENT_ENDTX);
+	nrf_spim_event_clear(spim, NRF_SPIM_EVENT_STARTED);
+}
+
+static void uarte_configuration_store(NRF_UARTE_Type *uarte,
```

Comment:
Why do you need to store this information and then restore when UART0 is activated again? Does the UART0 configuration register gets reset?

Reply:
Unfortunately UART and SPI share most registers. So when you want to configure SPI then you overwrite register values for UART and also if you left UART registers configured they values may have impact on the SPI behaviour.